### PR TITLE
Added 'mafia clean'

### PR DESCRIPTION
**Pull request by jacobstanley**
This calls cabal clean and also deletes the sandbox

### CODE REVIEW

**Comment by tranma** [comment]
more like 'mafia nuke' 😄
:+1:

**Comment by jacobstanley** [comment]
I'm reserving 'mafia nuke' for when you want to delete your global cache

**Comment by tranma** [comment]
'mafia deathstar' for reinstalling ghc?

**Comment by jacobstanley** [comment]
mafia defenestrate

File: test/Test/IO/Mafia/Chaos.hs [hunk 1]
```diff
@@ -72,7 +74,8 @@ data Profiling =
     deriving (Eq, Ord, Show)
 
 data Action =
-    Build Profiling
+    Clean
```
Comment (charleso) [inline test/Test/IO/Mafia/Chaos.hs:77]: Missing from `genAction`?

File: src/Mafia/Cabal/Sandbox.hs [hunk 2]
```diff
@@ -62,17 +61,42 @@ dropLeft :: Either a b -> Maybe b
...
 
+removeSandbox :: EitherT CabalError IO ()
+removeSandbox = do
+  -- remove sandbox directory and sandbox config file
+  dir <- getInferredSandboxDir
+  ignoreIO $ removeDirectoryRecursive dir
+  ignoreIO $ removeFile defaultConfig
+
+  -- remove root sandbox directory if it is now empty
+  whenM (doesDirectoryExist sandboxRoot) . ignoreIO $ do
```
Comment (thumphries) [inline src/Mafia/Cabal/Sandbox.hs]: why ignore? wouldn't we want to throw left / explode if this went wrong? likewise the two above

**Comment by charleso** [comment]
:+1:

File: src/Mafia/Cabal/Sandbox.hs (hunk 2 above)
Comment (jacobstanley) [inline src/Mafia/Cabal/Sandbox.hs]: It might get removed by something else after the directory exists check

Comment (jacobstanley) [inline src/Mafia/Cabal/Sandbox.hs]: Maybe I shouldn't bother protecting against that kind of thing here, just fail it happens

Comment (thumphries) [inline src/Mafia/Cabal/Sandbox.hs]: should probably handle DNE explicitly, lots of weird IOExceptions out there that we should yell about

File: test/Test/IO/Mafia/Chaos.hs (hunk 1 above)
Comment (jacobstanley) [inline test/Test/IO/Mafia/Chaos.hs:77]: good catch, fixed

File: src/Mafia/Cabal/Sandbox.hs (hunk 2 above)
Comment (jacobstanley) [inline src/Mafia/Cabal/Sandbox.hs]: I might just make this whole function assume a single thread, the rest of cabal probably can't handle multiple builds at the same time anyway

Comment (thumphries) [inline src/Mafia/Cabal/Sandbox.hs]: :+1:

**Pull request merged**